src/store/ducks/opensrp/hierarchies/utils.tsx

```diff
@@ -300,27 +296,70 @@ export const preOrderStructureCountComputation = (node: TreeNode) => {
   return thisNodesSelectedStructs;
 };
 
+/** helper util that helps add metaData from the metaData state slice to the respective node
+ * that owns the metaData
+ * @param nodes - a node or a list of nodes to apply the metadata to
+ * @param meta - meta object by jurisdiction
+ */
+export const applyMeta = (nodes: TreeNode[] | TreeNode | undefined, meta: Dictionary<Meta>) => {
+  if (!nodes) {
+    return [];
+  }
+  const localNodes = Array.isArray(nodes) ? [...nodes] : [nodes];
+  for (let i = 0, len = localNodes.length; i < len; i++) {
+    const node = localNodes[i];
+    const thisNodesMeta = Object.assign(meta[node.model.id] || {}, {
+      META_STRUCTURE_COUNT: node.model.meta[META_STRUCTURE_COUNT],
+    });
+    thisNodesMeta[META_STRUCTURE_COUNT] = node.model.meta[META_STRUCTURE_COUNT];
+    node.model.meta = thisNodesMeta;
+  }
+  return localNodes;
+};
+
 /** find the parent node in a tree given the parentId
  * @param tree - the tree
  * @param parentId - id of the parent node
+ * @param metaData - the tree's metaData - used to know which nodes to selected
  */
-export const findAParentNode = (tree: TreeNode | undefined, parentId: string | undefined) => {
+export const findAParentNode = (
+  tree: TreeNode | undefined,
+  parentId: string | undefined,
```

Comment:
Should we change this to `tree?` and `parentId?`?

Reply:
yeah